### PR TITLE
Refactor Codex ACP adapter into parser/stream/protocol modules

### PR DESCRIPTION
**Pull request by martin-purplefish**
## Summary
- split the Codex ACP adapter monolith into focused modules:
  - `command-metadata.ts` for shell command tokenization, chain splitting, display metadata, and approval scope keys
  - `stream-event-handler.ts` for Codex notification + replay streaming logic
  - `protocol-permission-handler.ts` for Codex server request / ACP permission bridging
  - `adapter-state.ts` + `acp-adapter-utils.ts` for shared internal types/helpers
- reduced `codex-app-server-acp-adapter.ts` from 3372 lines to 1871 lines by turning it into a composition/orchestration entrypoint
- split parser-focused coverage out of the monolithic adapter test into module-level tests:
  - `command-metadata.test.ts`
  - `stream-event-handler.test.ts`
  - `protocol-permission-handler.test.ts`
- removed duplicated parser cases from `codex-app-server-acp-adapter.test.ts` and kept adapter-level behavior coverage

## Why
This makes the ACP adapter easier to reason about and test in isolation by separating shell parsing, streaming event translation, and protocol/permission handling concerns.

## Validation
- `pnpm typecheck`
- `pnpm test -- src/backend/domains/session/acp/codex-app-server-adapter/command-metadata.test.ts src/backend/domains/session/acp/codex-app-server-adapter/protocol-permission-handler.test.ts src/backend/domains/session/acp/codex-app-server-adapter/stream-event-handler.test.ts src/backend/domains/session/acp/codex-app-server-adapter/codex-app-server-acp-adapter.test.ts`

---

> [!NOTE]
> **Medium Risk**
> Moderate risk: large refactor of Codex ACP adapter notification/permission handling could subtly change session update ordering or edge-case behavior. Also alters workspace creation/init to persist and apply a `startupModePreset`, affecting default session startup behavior.
> 
> **Overview**
> Refactors the Codex ACP adapter by extracting shared types/helpers (`adapter-state.ts`, `acp-adapter-utils.ts`), command parsing/approval scoping (`command-metadata.ts`), Codex notification replay/stream translation (`stream-event-handler.ts`), and server permission bridging (`protocol-permission-handler.ts`), leaving `codex-app-server-acp-adapter.ts` as a composition/orchestration entrypoint.
> 
> Updates tests accordingly by moving command-metadata cases into module-level suites and adding focused coverage for the new stream and permission handlers.
> 
> Adds a workspace `startupModePreset` (`non_interactive`/`plan`) that flows from the client create-workspace form through TRPC and creation metadata into workspace init, where it controls how the default agent session is started; also filters archiving workspaces out of project summary results and avoids a potential init cleanup race by awaiting the eager session-start promise on failure.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ba8a664993ca89f9432009e38cf367c6d247162c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/backend/domains/session/acp/codex-app-server-adapter/stream-event-handler.ts [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+      }
+      const path = asString(change.path);
+      if (!path) {
+        continue;
+      }
+      locations.push({ path });
+    }
+
+    return locations;
+  }
```
[the remark this answers is not in the document]
Comment (martin-purplefish) [reply src/backend/domains/session/acp/codex-app-server-adapter/stream-event-handler.ts]: @cursor - fixed as suggested. I extracted `extractLocations` into `acp-adapter-utils.ts` and both `stream-event-handler.ts` and `codex-app-server-acp-adapter.ts` now import the shared helper from there (commit ba8a6649).

**Comment by martin-purplefish** [comment]
@cursor please re-review
